src/test/java/redis/clients/jedis/authentication/TokenBasedAuthenticationIntegrationTests.java

```diff
@@ -0,0 +1,256 @@
+package redis.clients.jedis.authentication;
+
+import static org.mockito.Mockito.when;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.atLeast;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.awaitility.Awaitility.await;
+import static org.awaitility.Durations.ONE_HUNDRED_MILLISECONDS;
+import static org.awaitility.Durations.ONE_SECOND;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import redis.clients.authentication.core.IdentityProvider;
+import redis.clients.authentication.core.IdentityProviderConfig;
+import redis.clients.authentication.core.SimpleToken;
+import redis.clients.authentication.core.TokenAuthConfig;
+import redis.clients.jedis.CommandArguments;
+import redis.clients.jedis.Connection;
+/*  */
+import redis.clients.jedis.DefaultJedisClientConfig;
+import redis.clients.jedis.EndpointConfig;
+import redis.clients.jedis.HostAndPorts;
+import redis.clients.jedis.JedisClientConfig;
+import redis.clients.jedis.JedisPooled;
+import redis.clients.jedis.JedisPubSub;
+import redis.clients.jedis.RedisProtocol;
+import redis.clients.jedis.Protocol.Command;
+import redis.clients.jedis.exceptions.JedisException;
+
+public class TokenBasedAuthenticationIntegrationTests {
+  private static final Logger log = LoggerFactory
+      .getLogger(TokenBasedAuthenticationIntegrationTests.class);
+
+  private static EndpointConfig endpointConfig;
+
+  @BeforeClass
+  public static void before() {
+    try {
+      endpointConfig = HostAndPorts.getRedisEndpoint("standalone0");
+    } catch (IllegalArgumentException e) {
+      try {
+        endpointConfig = HostAndPorts.getRedisEndpoint("standalone");
```

Comment:
I'll fix the endpoint name; there's no need for workarounds here.

Reply:
so we will use "standalone0" instead of "standalone", right ?

Reply:
yes